hyperspy/_signals/eds_tem.py

```diff
@@ -258,28 +259,32 @@ def get_calibration_from(self, ref, nb_pix=1):
             mp.Acquisition_instrument.TEM.Detector.EDS.live_time = \
                 mp_ref.Detector.EDS.live_time / nb_pix
 
+
     def quantification(self,
-                       intensities,
-                       kfactors,
-                       composition_units='weight',
+                       intensities='auto',
+                       method='auto',
```

Comment:
if you make this just 'method', rather than 'method='auto'' then the function will tell you that you haven't specified enough stuff unless you specify method. That would allow you to remove the else and exception raising, so would be a little more code efficient and I think preferable.